akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -88,7 +88,7 @@ private[http] class HttpRequestRendererFactory(
             renderHeaders(tail, hostHeaderSeen, userAgentSeen, transferEncodingSeen)
 
           case x: RawHeader if (x is "content-type") || (x is "content-length") || (x is "transfer-encoding") ||
-            (x is "host") || (x is "user-agent") ⇒
```

Comment:
This is not enough. You also need to adapt `userAgentSeen`, otherwise, the default user agent will be added as well.